controller.go

```diff
@@ -268,7 +268,7 @@ func (c *controller) SetClusterProvider(provider cluster.Provider) {
 	}
 	c.Unlock()
 
-	if provider == nil || sameProvider {
+	if provider == nil || sameProvider && c.getAgent() != nil {
```

Comment:
If you can use "provider == nil || ( sameProvider && c.getAgent() != nil  )" - will be more user readable